force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -354,15 +470,29 @@ public with sharing class FlowControlLoopsCollections {
         // add Jane Smith => 60000
         // add Sam Brown => 55000
         // add Alice Johnson => 65000
+        employeeSalaries.put('John Doe', 50000);
+        employeeSalaries.put('Jane Smith', 60000);
+        employeeSalaries.put('Sam Brown', 55000);
+        eemployeeSalaries.put('Alice Johnson', 65000);
+
 
         // Initialize a list to store the names of high paid employees
+        List<string> highPaidEmployees = new List<string>();
 
         // Iterate over the Map using a for loop
+        for (Map.Entry<String, Integer> entry : employeeSalaries.entrySet()) {
+            // get the name and salary of the employee
+            String  name = entry.getKey();
+            Integer salary = entry.getValue();
             // Check if the salary of the employee is more than 55000
+            if (entry.getValue() > 55000) {
                 // Add the employee to the list of high paid employees
+                highPaidEmployees.add(entry.getKey());
+        }
+    }
```

Comment:
Keep working on this. You already have your Map.  In your loop you want to iterate over the KeySet. 

https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_map.htm